src/process/ProcessManagerImpl.cpp

```diff
@@ -235,13 +233,18 @@ ProcessManagerImpl::tryProcessShutdown(std::shared_ptr<ProcessExitEvent> pe)
         auto runningIt = mProcesses.find(pid);
         if (runningIt != mProcesses.end())
         {
-            // Mark it as "ready to be killed"
+            // Mark it as "ready to be killed" if caller retries.
```

Comment:
note: `mKillable` (and `mPending`) do enforce uniqueness of elements, maybe we should just make them a map instead of a deque

Reply:
Yeah, I thought of moving to a map/set instead but held off. The deque is (as far as I know) there to enforce ordering on operations -- so that we kill processes (and thereby start subsequent processes) in as-close-to-ordered as possible, rather than random pointer order.

I'm not sure this _really_ helps though since processes are likely to complete and exit in an arbitrary (or at least not-totally-predictable) order. And since we ultimately respond to SIGCHLDs as they come in, maybe we'll be pulling stuff out in effectively-random order anyway.

Reply:
Pushed version that uses a set (and in fact two sets -- one for "we sent SIGTERM" and one for "we sent SIGKILL"). Kept the pending deque a deque because it _is_ relevant which order we run processes in, even if they exit in a random order.